### PR TITLE
View trainee process for supervisor

### PR DESCRIPTION
**Pull request by luongvietdung**

### CODE REVIEW

File: app/views/supervisor/users/_trainee_progress.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<% if @user.courses.count > 0 %>
+  <% @user.courses.active.each do |course| %>
+    <%= course.name%>
```
Comment (trinhducduy) [inline app/views/supervisor/users/_trainee_progress.html.erb]: add space

File: app/views/supervisor/users/index.html.erb [hunk 2]
```diff
@@ -19,6 +20,9 @@
         <td><%= user.name %></td>
         <td><%= user.email %></td>
         <td>
+          <%= link_to "View progess", supervisor_user_path(user), class: "btn btn-info", data: {toggle:"modal", target:"#traineeProgressModal", remote: true, method: :get} %>
```
Comment (trinhducduy) [inline app/views/supervisor/users/index.html.erb]: i18n

**Comment by trinhducduy** [comment]
ok

**Comment by trungnn142** [comment]
ok

File: app/models/course.rb [hunk 3]
```diff
@@ -43,6 +43,11 @@ def finished? user
 
   def owned_by? user
     !course_users.detect{|course_user| course_user.user_id == user.id &&
-      course_user.is_owner == true}
+      course_user.is_owner == true}.nil?
```
Comment (ThuBM) [inline app/models/course.rb]: remove .nil?

File: app/views/supervisor/users/_trainee_progress.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<% if @user.courses.count > 0 %>
+  <% @user.courses.active.each do |course| %>
+    <%= course.name %>
+    <div class="progress">
+      <div class="progress-bar" role="progressbar"
+        style="width:<%= number_to_percentage course.percent(@user), precision: 0 %>">
```
Comment (ThuBM) [inline app/views/supervisor/users/_trainee_progress.html.erb]: style viết trong css

File: app/views/supervisor/users/index.html.erb [hunk 5]
```diff
@@ -19,6 +20,11 @@
         <td><%= user.name %></td>
         <td><%= user.email %></td>
         <td>
+          <%= link_to t("application.buttons.view_progress"), supervisor_user_path(user),
+            class: "btn btn-info", data: {toggle:"modal", target:"#traineeProgressModal",
```
Comment (ThuBM) [inline app/views/supervisor/users/index.html.erb]: add space after target:

File: app/models/user.rb [hunk 6]
```diff
@@ -51,7 +51,7 @@ def authenticated? remember_token
 
   private
   def default_password
-    self.password = "Tms32@2015" if self.trainee?
+    self.password = "Tms32@2015" if self.trainee? && self.password.nil?
```
Comment (ThuBM) [inline app/models/user.rb]: remove .nil?

**Comment by ThuBM** [comment]
OK

**Pull request merged**